src/battle_animation.cpp

```diff
@@ -203,7 +203,7 @@ void BattleAnimation::UpdateFlashGeneric(int timing_idx, int& r, int& g, int& b,
 void BattleAnimation::UpdateScreenFlash() {
 	int r, g, b, p;
 	UpdateFlashGeneric(screen_flash_timing, r, g, b, p);
-	Main_Data::game_screen->FlashOnce(r, g, b, p, 0);
+	if (r > 0 || g > 0 || b > 0 || p > 0) Main_Data::game_screen->FlashOnce(r, g, b, p, 0);
```

Comment:
This needs further investigation.

Battle animations when run actually cancel screen flash effect each frame. 

Try an event like this on the map

```
FlashScreen(31,31,31,31,12sec)
Wait 1.0sec
Show BattleAnimation on player
```

This happens for map and battle animations.

For these battler animations, probably different update logic is used to prevent this. I haven't deep dived into battler animations in RE tools yet as it's super complicated and kind of the final frontier of the 2k3 battle system.

If I get some time I can try to look more how this works.

Reply:
This is hard to explain but it somehow looks like that battle animation sprites are "blocking" the flash functions. This `FlashOnce` is called every frame with all parameters set to 0. Adding this check above fixes the flashing in RPG Maker 2003 battles with battle animation sprites used but as you said this might break the flash functions in other situations.

Reply:
> that battle animation sprites are "blocking" the flash functions.

Yes, and thats what I was saying. This is on purpose. I actually did some tests where I saved the game each frame and as soon as the battle animation starts you see it clobbering the flash screen chunks. It also works this way for map character flash chunks.

These battler battle animations probably work differently. So your proposed patch of filtering here would only be valid for battler battle animations, but not regular battle and map animations.